tools/walletextension/rpcapi/utils.go

```diff
@@ -67,11 +71,16 @@ func UnauthenticatedTenRPCCall[R any](ctx context.Context, w *Services, cfg *Cac
 		return withPlainRPCConnection(w, func(client *rpc.Client) (*R, error) {
 			var resp *R
 			var err error
+
+			basectx := ctx
```

Comment:
Maybe we could just return an error if context is nil (if there's no way to enforce not nil). I had a quick glance and it seems like all usages pass in a context.

Reply:
good idea. simplifies it